crates/networking/p2p/net.rs

```diff
@@ -768,19 +773,16 @@ async fn handle_peer_as_initiator(
         .connect(SocketAddr::new(node.ip, node.tcp_port))
         .await
         .unwrap();
-    let conn = RLPxConnection::initiator(signer, msg, stream).await;
+    let conn = RLPxConnection::initiator(signer, msg, stream, storage).await;
     handle_peer(conn, table).await;
 }
 
 async fn handle_peer(mut conn: RLPxConnection<TcpStream>, table: Arc<Mutex<KademliaTable>>) {
     match conn.handshake().await {
-        Ok(_) => {
-            // TODO Properly build listen loop
-            // https://github.com/lambdaclass/lambda_ethereum_rust/issues/840
-            // loop {
-            //     conn.await_messages();
-            // }
-        }
+        Ok(_) => match conn.handle_peer().await {
+            Ok(_) => unreachable!(),
```

Comment:
why is this unreachable?

Reply:
In current state, `handle_peer` sets an infinite loop that breaks only on error.

It may change in the future if we have a _graceful_ way to end the loop.